tools/challenge-editor/client/src/components/editor/editor.tsx

```diff
@@ -53,11 +57,11 @@ const Editor = () => {
   };
 
   const handleChange = (
-    editor: codemirror.Editor,
-    data: codemirror.EditorChange,
-    value: string
+    instance: codemirror.Editor,
+    change: codemirror.EditorChange
   ) => {
-    setInput(value);
+    setInput(change.text.join(' '));
```

Comment:
Currently, with this logic, it looks like this, but works

![image](https://user-images.githubusercontent.com/88248797/218070565-d4e6c2aa-5e2c-4e7e-8af0-76ad3b057aa7.png)

Without it, looks like this but doesn't

[rec-tab.webm](https://user-images.githubusercontent.com/88248797/218070699-78ba7e07-8c21-485f-adc2-5d7a35c61f38.webm)

Reply:
Oh it doesn't show the message, but the message say the change did happen, but the change doesn't happen.

And the old logic I mean 

```ts
 const handleChange = (
    editor: codemirror.Editor,
    data: codemirror.EditorChange,
    value: string
  ) => {
    setInput(value);
```